src/js/graph/node.js

```diff
@@ -76,6 +79,130 @@ Node.prototype.hasTraitSpanContainer = function() {
 Node.prototype.setGraph = function(graph) {
     this._graph = graph;
 };
+Node.prototype.removeGraph = function() { this._graph = null; };
+Node.prototype.connectEdges = function() {
+    if (!this._graph) { return; }
+    var self = this, graphImpl = this._graph._graphImpl;
+    self.getEdges().forEach(function(edge) {
+        // Check if the edge exists before creating it.
+        if (graphImpl.edge(self.getId(), edge.getTargetId()) === edge.getEdgeType()) { return; }
+        self._connectEdgesSpanContainer();
+        self._connectEdgesSequence();
+        // console.log("LINK EDGE nodeId: " + self.getId() + ", targetId: " + edge.getTargetId() + ", type: " + edge.getEdgeType() + ".");
+        graphImpl.setEdge(self.getId(), edge.getTargetId(), edge.getEdgeType());
+    });
+};
+// If node is a span container create edges to all it's children.
+// This allows children to quickly reference their parent nodes.
+// Child nodes may not have edges in the graphImpl yet so create edges
+// based only on the edges array in the node.
+Node.prototype._connectEdgesSpanContainer = function() {
+    var graph = this._graph, graphImpl = graph._graphImpl;
+    var firstEdge = this.getFirstEdgeByType('first');
+    var lastEdge = this.getFirstEdgeByType('last');
+    if (this.hasTraitSpanContainer() && firstEdge && lastEdge) {
+        var linkNode = graph.getNodeById(firstEdge.getTargetId());
+        var lastNode = graph.getNodeById(lastEdge.getTargetId());
+        while (true) {
+            // console.log("Node._connectEdgesSpanContainer: Checking node '" + linkNode.getId() + "', type: '" + linkNode.getType() + "'.");
+            // Don't create the edge if it already exists
+            if (!graphImpl.edge(this.getId(), linkNode.getId())) {
+                // console.log("LINK EDGE nodeId: " + this.getId() + ", targetId: " + linkNode.getId() + ", type: child.");
+                graphImpl.setEdge(this.getId(), linkNode.getId(), 'child');
+            }
+            if (linkNode === lastNode) {
+                // console.log("Node._connectEdgesSpanContainer: LinkNode === LastNode.");
+                break;
+            }
+            try {
+                linkNode = graph.getNodeById(linkNode.getFirstEdgeByType('next').getTargetId());
+            } catch (err) {
+                // Something went wrong, maybe this sequence is messed up?
+                // * Doesn't have a next edge.
+                // * Last edge is incorrect.
+                // console.error("There was an issue linking children for node '" + this.getId() + "'.");
+                break;
+            }
+        }
+    }
+};
+Node.prototype._connectEdgesSequence = function() {
+    var self = this;
+    if (!self.hasTraitSequence()) { return; }
+    // Sequence (first)
+    // Handled by the generic edge loop above.
+    // Sequence (middle)
+    // A -> B
+    // A C -> B
+    // A -> C -> B
+    if (self.hasNext() && !self.hasPrevious() &&
```

Comment:
I believe the comments are clear enough here and since this block of code is not being used elsewhere I don't see a need to push it into it's own function. If more functionality falls into this function in the future then we can break them out.